tests/Integration/Momento.Sdk.Tests/VectorIndexDataTest.cs

```diff
@@ -0,0 +1,333 @@
+public class VectorIndexDataTest : IClassFixture<VectorIndexClientFixture>
+{
+    private readonly IPreviewVectorIndexClient vectorIndexClient;
+
+    public VectorIndexDataTest(VectorIndexClientFixture vectorIndexFixture)
+    {
+        vectorIndexClient = vectorIndexFixture.Client;
+    }
+
+    [Fact]
+    public async Task UpsertItemBatchAsync_InvalidIndexName()
+    {
+        var response = await vectorIndexClient.UpsertItemBatchAsync(null!, new List<Item>());
+        Assert.True(response is UpsertItemBatchResponse.Error, $"Unexpected response: {response}");
+        var error = (UpsertItemBatchResponse.Error)response;
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, error.InnerException.ErrorCode);
+
+        response = await vectorIndexClient.UpsertItemBatchAsync("", new List<Item>());
+        Assert.True(response is UpsertItemBatchResponse.Error, $"Unexpected response: {response}");
+        error = (UpsertItemBatchResponse.Error)response;
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, error.InnerException.ErrorCode);
+    }
+
+    [Fact]
+    public async Task DeleteItemBatchAsync_InvalidIndexName()
+    {
+        var response = await vectorIndexClient.DeleteItemBatchAsync(null!, new List<string>());
+        Assert.True(response is DeleteItemBatchResponse.Error, $"Unexpected response: {response}");
+        var error = (DeleteItemBatchResponse.Error)response;
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, error.InnerException.ErrorCode);
+
+        response = await vectorIndexClient.DeleteItemBatchAsync("", new List<string>());
+        Assert.True(response is DeleteItemBatchResponse.Error, $"Unexpected response: {response}");
+        error = (DeleteItemBatchResponse.Error)response;
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, error.InnerException.ErrorCode);
+    }
+
+    [Fact]
+    public async Task SearchAsync_InvalidIndexName()
+    {
+        var response = await vectorIndexClient.SearchAsync(null!, new List<float> { 1.0f });
+        Assert.True(response is SearchResponse.Error, $"Unexpected response: {response}");
+        var error = (SearchResponse.Error)response;
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, error.InnerException.ErrorCode);
+
+        response = await vectorIndexClient.SearchAsync("", new List<float> { 1.0f });
+        Assert.True(response is SearchResponse.Error, $"Unexpected response: {response}");
+        error = (SearchResponse.Error)response;
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, error.InnerException.ErrorCode);
+    }
+
+    [Fact]
+    public async Task UpsertAndSearch_InnerProduct()
+    {
+        var indexName = $"index-{Utils.NewGuidString()}";
+
+        var createResponse = await vectorIndexClient.CreateIndexAsync(indexName, 2, SimilarityMetric.InnerProduct);
+        Assert.True(createResponse is CreateIndexResponse.Success, $"Unexpected response: {createResponse}");
+
+        try
```

Comment:
This repeated code and cleanup can be refactored to a disposable class, then you can use a using block; see https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/using